services/QuillLMS/app/models/question.rb

```diff
@@ -168,6 +160,46 @@ def connect_sentence_combining?
   # Translatable
   def self.translatable_field_name = 'instructions'
 
+  private def add_data_for(type:, new_data:)
+    if stored_as_array?(type)
+      id = data[type].length
+    else
+      id = new_uuid
+    end
+    return id if set_data_for(type:, id:, new_data:)
+  end
+
+  private def add_uid(data_array:)
```

Comment:
Maybe not worth doing anything about, but naming this `data_array` when it's not always an array (as evidenced by the next line) confused me a little. Maybe just `data`?